composition-js/src/merging/merge.ts

```diff
@@ -1906,11 +1914,33 @@ class Merger {
     return source.locations.filter(loc => isExecutableDirectiveLocation(loc));
   }
 
+  // In general, we want to merge applied directives after merging elements, the one exception
+  // is @inaccessible, which is necessary to exist in the supergraph for EnumValues to properly
+  // determine whether the fact that a value is both input / output will matter
```

Comment:
Nit: the naming of that method, especially along side the almost similar `mergeAllAppliedDirectives`, gets maybe a little bit uninformative on first approximation. It's all private so not a huge deal, but maybe this one can be renamed to something longer but more informative (say `recordAppliedDirectivesToMerge`? of course, it does merge `@inaccessible` in practice, but it's more a special case so maybe ok if not reflected in the name).